scripts/pylint_extensions.py

```diff
@@ -795,6 +795,75 @@ def visit_functiondef(self, node):
             self.add_message('function-args-order-cls', node=node)
 
 
+class RestrictedImportChecker(checkers.BaseChecker):
+    """Custom pylint checker which checks layers importing modules
+    from their respective restricted layers.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+    name = 'invalid-import'
+    priority = -1
+    msgs = {
+        'C0009': (
+            'Importing %s layer in %s layer is prohibited.',
+            'invalid-import',
+            'Importing some layers in some specific layers is '
```

Comment:
This is unclear. Please see messages for other checks.

Reply:
Hi, @apb7 Thanks for the review. I have updated the message PTAL and let me know if it needs more improvement.
Thanks!